ros2topic/ros2topic/verb/pub.py

```diff
@@ -120,17 +124,20 @@ def main(args):
     if args.once:
         times = 1
 
-    if args.stdin:
-        values = collect_stdin()
-    else:
-        values = args.values
+    values = None
+    if not args.yaml_file:
+        if args.stdin:
+            values = collect_stdin()
+        else:
+            values = args.values
```

Comment:
I feel like this leads to a lot of `if` statements in the rest of the code.  Is there a way we could make `values` contain either the value from `stdin`, the value passed in `args.values`, or from the yaml file?  In other words, the rest of the code would be much nicer if this looked like:

```suggestion
    values = None
    if args.yaml_file:
        values = get_values_from_yaml_file(args.yaml_file)
    else:
        if args.stdin:
            values = collect_stdin()
        else:
            values = args.values
```

Reply:
because of https://github.com/ros2/ros2cli/pull/925/commits/853e7f2720510c1e0279a1754d4fe0fd61837971, this additional if statement is no longer needed.